test/operators/second_quantization/test_fermionic_op.py

```diff
@@ -141,6 +141,16 @@ def test_init_from_tuple_label(self):
         ]
         self.assertListEqual(actual.to_list(), desired)
 
+    def test_iter(self):
+        """Test __iter__"""
+        op = FermionicOp([("+", 1.0), ("N", 2.0), ("-+", 3.0)], display_format="dense")
+        expected = [
+            ([("+", 0)], 1.0),
+            ([("+", 0), ("-", 0)], 2.0),
+            ([("-", 0), ("+", 1)], 3.0),
+        ]
+        self.assertEqual(list(op), expected)
```

Comment:
The unit tests tests the public API. In practice you would not normally call on list on it since it removes any advantage of an iterator. In this case since its all in memory anyway though it really is no different. I just think it clearer when it states it is going to test iter that you call `op.iter()` in the unit test. You may find it easier to write the test after coercing it to a list, that's fine.

Reply:
The method is called `__iter__`, not `iter`. In practice, you would never call that directly. It's a hidden dunder method. Most commonly you would do
```python
for term, coeff in op:
    ...
```

Reply:
We have people at different levels of Python understanding - I just think making it explicit can be clearer that its obviously testing what it says - the code doing list cast in Python no doubt checks for that built-in and can use that implicitly. It maybe clear to you what happening but I just think its better to be explicit in the unit test there rather than relying on implicit behavior.

Reply:
`list(iter(op))` is not idiomatic Python. We should strive to write idiomatic Python, even in the tests. If the reader doesn't know what `list` does, they can easily look it up.